gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -103,6 +103,7 @@ def self._check_final_ancestors(target, target_ancestors, source_method_names)
     # use reverse_each to check farther-up ancestors first, for better error messages. we could avoid this if we were on
     # the version of ruby that adds the optional argument to method_defined? that allows you to exclude ancestors.
     target_ancestors.reverse_each do |ancestor|
+      next if !module_with_final?(ancestor)
```

Comment:
Does this make the `!module_with_final?` check from ~6 lines above redundant?

Reply:
I don't think so.  The one above is a fast reject for when the module has never ever seen `final` anywhere, either in itself, or in its ancestor chain.  The one being added here is when something has added some bit of a`final` thing to `target` -- we don't know what -- and we're rejecting things that obviously can't be responsible for an addition of `final`.

Come to think of it, that suggests another optimization we could be doing: right now, we're eagerly computing `target_ancestors` and `source_method_names`, but both of those are relatively expensive, especially in the case of an included-esque module: we're set differencing ancestor chains and computing all instance methods on the included-ish module.  Maybe if we saved both of those until we knew we actually needed them, we could save cycles?